### ISSUE TITLE
Request: Splash Quads, Waterfall Particles

### ISSUE THREAD

**Issue by Lafolie** [Goby56/wakes#13]
Hello again, really love this mod! I would like to request a couple of features that would make this the *Ultimate Water FX Mod<sup>TM</sup>*. They are both features from the now Quilt-only mod "Effective". Effective is ARR, so they can't be ported but, seeing how fantastic the waves are, I believe you could do a better job yourself.

### Splash Quads
These are 4 quads in a cube arrangement (just the sides) that act as '3D' splashes that pop out of the water when something falls in. They can be scaled by velocity, and not triggered at all if the velocity magnitude is too low. Effective uses 2 sets of these, with an inner and outer splash. This effect may be similar to your upcoming boat splashes feature.

### Waterfall Particles
These are just misty-looking particles that are formed wherever water flows downwards into another source. Potentially tricky to get right; I believe effective has some sort of threshold that the liquid has to travel before it counts as a waterfall (I recall seeing some false positives too). They also have a waterfall sound that plays, which is nice, but it was quite poorly executed and the sample they used didn't loop well at all.

---
Effective also has bubbles that raise out of things opened when under water, though they seem to be more of a gameplay thing than just a visual effect.

I hope this isn't too much to ask! Wakes is by far my favourite mod, and I would love to see it perfected as the ultimate water fx mod :)

P.S. Have you written about the boat splashes anywhere? I'm excited for it, and was wondering what your vision for it is? Something like Wind Waker perhaps...?

**Comment by Lafolie** [comment]
No worries, I love hitting F5 in the boats, hehe. I recently switched my worldgen to generate continents, so I will be boating a lot!

> If I would implement any of their effects I feel like I should give it my own take to avoid making it into a clone or "effective for fabric".

Yeah this was my sentiment, sorry if that didn't come across properly in my last comment. Effective is great and I agree that you shouldn't just rip it off, but I bet there's a cooler way to pull off a similar effect. Besides, there is far more to Effective than the splashes (their Allay trails are really cool).

Submersion splashes would work well for boats anyway - they could be triggered when a boat falls from a height (such as from a waterfall). Perhaps they could be angled outwards, and/or you could use a radial arrangement (hexagonal?). If angled, the sine of the angle could interpolate from 1 to 0, giving the water a weighty feel.

If you really wanted to go the extra (nautical) mile, as you have with the current wave nodes, you could have any kind of splash (be it the planned frontal ones, or the ones I suggested) interact. Splashes could merge and become more foamy etc. Maybe splashes could even trigger wave nodes!

I say all this without your vision though, just throwing it out there.

**Comment by Goby56** [comment]
Yeah angled splashes would be really cool and I especially like your idea of interpolated angles. I didn't have that in mind but I think that would give the mod its own kind of characteristic and help it stand out.

**Comment by Lafolie** [comment]
Awesome stuff. Adding that as-is should be straightforward, but there is a small design consideration: if you use 4 quad planes, as the angle changes the planes will break apart as the edges are no longer axis-aligned. This could look bad, hard to say without actually seing it.

Possible solution is to have additional tris that bridge the gap formed by the 'unboxing' of the quads.

**Comment by tjhaskel** [comment]
Seconding this, Effective can't currently be installed on quilt 1.20.4 due to quilted fabric API being out of date. Wakes is a fabulous alternative but I miss the waterfall splashes. Thank you for making such a nice mod!

**Comment by souls-are-tasty** [comment]
> When it comes to the effects you stated I don't really know how I feel about ripping of their work. As stated before, I will implement splashes due to horizontal velocity but vertical, I'm not so sure. 
> 
> If I would implement any of their effects I feel like I should give it my own take to avoid making it into a clone or "effective for fabric".

I wouldn't rlly say you'd be ripping them off as the splash was based off the splashes from the minecraft 1.18 trailer. But I'm nevertheless happy you're making your own creative take on minecraft splashes

**Comment by Goby56** [comment]
I'm going to have to close this issue as I think Wakes should do one thing: wakes. Effective already does the mentioned effects really well and it has been ported to fabric if I recall correctly so there is no reason to implement this. The angled splashes does sound interesting but for now this is not high up on the list of to-do:s.
